docs/section6/Rule6-2.md

```diff
@@ -14,49 +14,47 @@
 **Manual Check:** None  
 **Evaluation Context:** Each Data Element  
 **Data Lookup:** Table G3.7 and Table G3.8  
-## Rule Logic: 
-
-- For each building_segment in the Proposed Model: ```building_segment_proposed in P_RMR.building.building_segments```  
+**Function Call:** None
 
-- Get the lighting_building_area_type for the building_segment: ```lighting_building_area_type_proposed = building_segment_proposed.lighting_building_area_type```  
+## Rule Logic: 
 
-  - Determine if the building segment uses Building Area Method: ```if lighting_building_area_type_proposed is "None": bam_flag = FALSE else bam_flag = TRUE```  
+- For each building segment in the Proposed Model: `building_segment_p in P_RMR.building.building_segments`  
 
-- For each thermal_block in building_segment: ```thermal_block_proposed in building_segment_proposed.thermal_blocks:```  
+  - If building segment specifies lighting building area type, get the allowable lighting power density from Table G3-8: `if building_segment_p.lighting_building_area_type in table_G3_8: allowable_LPD_BAM = data_lookup(table_G3_8, building_segment_p.lighting_building_area_type)`  
 
-- For each zone in thermal_block: ```zone_proposed in thermal_block_proposed.zones:```  
+  - For each thermal block in building segment: `thermal_block_p in building_segment_p.thermal_blocks:`  
+  
+    - For each zone in thermal block: `zone_p in thermal_block_p.zones:`  
 
-- For each space in zone: ```space_proposed in zone_proposed.spaces:```  
+      - For each space in zone: `space_p in zone_p.spaces:`  
 
-  - Get floor_area from space: ```floor_area_proposed = space_proposed.floor_area```  
+        - Add space lighting power to building segment total: `design_lighting_wattage += space_p.interior_lighting.power_per_area * space_p.floor_area`  
 
-  - If bam_flag is TRUE then add floor_area to the total building segment floor area: ```total_building_segment_area_proposed += floor_area_proposed```  
+        - If building segment specifies lighting building area type , add space floor area to the total building segment floor area: `if allowable_LPD_BAM: total_building_segment_area_p += space_p.floor_area`  
 
-    - Get the allowable lighting power density for this building area type: ```if lighting_building_area_type_proposed in table_G3_8: allowable_LPD_proposed = data_lookup(table_G3_8, lighting_building_area_type_proposed) else raise_warning```  
+        - Check if any space does not specify lighting space type, flag for Building Area Method: `if NOT space_p.lighting_space_type in table_G3_7: BAM_flag = TRUE`  
 
```

Comment:
I think instead of BAM_flag=TRUE it should be SBS_flag=FALSE. Because missing space types means that we can't use space-by-space method. It doesn't automatically mean that we can use BAM because BAM area type may not be provided either.

Reply:
This flag is just to mark space-by-space method should not be used and does not mean the rule will use BAM method (as the rule assertion below uses both the BAM_flag and the allowable_LPD_BAM flag to determine BAM method). I would prefer to have a TRUE flag instead of FALSE. I can change to name to avoid confusion.

Reply:
Changing the name would work, but then would it be No_SBS_flag=TRUE? Ultimately the names don't matter, but if they are not intuitive the logic may be screwed up during the implementation.